server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocatorTests.java

```diff
@@ -269,6 +260,26 @@ public void testBalanceByActualAndForecastDiskUsage() {
         );
     }
 
+    public void testDoNotBalancePartialIndicesByDiskUsage() {
+
+        var allocationService = createAllocationService(WITH_DISK_BALANCING, () -> createClusterInfo(Map.of()));
+
+        var partialSearchableSnapshotSettings = indexSettings(IndexVersion.current(), 1, 0) //
+            .put(SETTING_IGNORE_DISK_WATERMARKS.getKey(), true);
+
+        var clusterState = applyStartedShardsUntilNoChange(
+            stateWithStartedIndices(
+                anIndex("frozen-index-1", partialSearchableSnapshotSettings).shardSizeInBytesForecast(ByteSizeValue.ofGb(1).getBytes()),
+                anIndex("frozen-index-2", partialSearchableSnapshotSettings).shardSizeInBytesForecast(ByteSizeValue.ofGb(1).getBytes()),
+                anIndex("frozen-index-3", partialSearchableSnapshotSettings).shardSizeInBytesForecast(ByteSizeValue.ofGb(1).getBytes()),
+                anIndex("frozen-index-4", partialSearchableSnapshotSettings).shardSizeInBytesForecast(ByteSizeValue.ofGb(10).getBytes())
+            ),
+            allocationService
+        );
+
+        assertThat(getShardsPerNode(clusterState).values(), everyItem(hasSize(2)));
```

Comment:
BTW, what are these two items? A comment may be helpful. My guess is that one is the with the zero searchable snapshots?

Reply:
All indices are partial searchable snapshots here. We do not care where each index is allocated as long as they are balanced by count across the nodes